pkg/git/providers/git/git.go

```diff
@@ -0,0 +1,48 @@
+package git
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/aws/eks-anywhere/pkg/logger"
+)
+
+const (
+	EksaGitPasswordTokenEnv   = "EKSA_GIT_PASSWORD"
+	EksaGitPrivateKeyTokenEnv = "EKSA_GIT_PRIVATE_KEY"
+)
+
+type Options struct {
+	RepositoryUrl string
+	User          string
+}
+
+func GetGitPasswordFromEnv() (string, error) {
+	password, err := ReadEnvVar(EksaGitPasswordTokenEnv)
+	if err != nil {
+		return "", err
+	}
+
+	return password, nil
+}
+
+func GetGitPrivateKeyFile() (string, error) {
+	privateKeyFilePath, err := ReadEnvVar(EksaGitPrivateKeyTokenEnv)
+	if err != nil {
+		return "", err
+	}
+
+	return privateKeyFilePath, nil
+}
+
+func ReadEnvVar(EksaEnvVar string) (string, error) {
```

Comment:
I think we should move this to a different package, it's not git specific

Also, what do you think about a different name that tells us that it will fail if the variable is not set or empty?
I'm out of good ideas but for example `ReadAndExpectEnvVar`